src/Juvix/Backends/Michelson/Compilation/Datatypes.hs

```diff
@@ -26,18 +27,35 @@ unpack ∷
 unpack (Type ty _) binds =
   case ty of
     Tbool → do
-      modify @"stack" (drop 1)
+      modify @"stack" (VStack.drop (1 ∷ Int))
       return (SeqEx [])
     TPair _ _ fT sT →
       case binds of
         [Just fst, Just snd] → do
-          modify @"stack" (appendDrop [(VarE fst, fT), (VarE snd, sT)])
+          modify @"stack"
+            ( VStack.appendDrop
+                ( VStack.fromList
+                    [ (VStack.varE fst Nothing, fT),
```

Comment:
Is it worth a helper function for `VarE x Nothing` ?

Reply:
Done